doc/index.rst

```diff
@@ -29,23 +34,21 @@ PennyLane-Lightning provides the following device:
 
    installation
    support
-   Architecture <arch/arch_overview>
 
 .. toctree::
    :maxdepth: 2
    :caption: Usage
    :hidden:
 
-   devices
+   lightning_qubit/device
+   lightning_kokkos/device
 
 .. toctree::
    :maxdepth: 2
    :caption: Development
    :hidden:
 
-   add_kernel
-   benchmark
-   avx_kernels/index
```

Comment:
@AmintorDusko this is very minor, but it might be nice to keep all the development pages in the sidebar, for visibility? E.g., you could have

```
.. toctree::
   :maxdepth: 2
   :caption: Development
   :hidden:

   lightning_qubit/development/add_gate_kernel

.. toctree::
   :maxdepth: 2
   :caption: AVX Kernels
   :hidden:

   lightning_qubit/avx_kernels/build_system
   lightning_qubit/avx_kernels/implementation
```

with the new file structure. 

I find it a bit hard to navigate otherwise 🤔

Reply:
We could have three possible categories of development pages.
One for `lightning.qubit` and one for `lightning.kokkos`, and maybe in the future some development pages that apply to both.  Any suggestion on how to handle it? I mean should we have a single development page? How should we signalize to which backend it applies?

Reply:
I think we should bring @trbromley int discuss this change.